week3/homework/src/index.js

```diff
@@ -1,3 +1,102 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+
+const express = require('express');
+const uuid = require('uuid/v4');
+
+const { readFile: _readFile, writeFile: _writeFile } = require('fs');
+
+const { promisify } = require('util');
+
+const readFile = promisify(_readFile);
+const writeFile = promisify(_writeFile);
+
+const app = express();
+
+// Use built-in JSON middleware to automatically parse JSON
+app.use(express.json());
+
+const TODO_FILE = 'todo.json';
+
+function readTodos() {
+  return readFile(TODO_FILE, 'utf8').then(
+    JSON.parse,
+    () => []
+  );
+}
+
+function writeTodos(todos) {
+  return writeFile(TODO_FILE, JSON.stringify(todos, null, 2));
+}
+
+// Read all todos
+app.get('/todos', async (req, res) => {
+  const todos = await readTodos();
+  res.json(todos);
+});
+
+// Read a todo by ID
+app.get('/todos/:id', async (req, res) => {
+  const id = req.params.id;
+  const todos = await readTodos();
+  if (id in todos) {
+    res.json(todos[id]);
```

Comment:
By default, you are using an array to store your todos (see line 25). `uuid` returns a `string` so your ids are going to look like `f64f2940`. This is not really how arrays are supposed to be used. Instead you should an object as your default todo store so that each property of that object will be an id. Does that make sense?